tests/rptest/tests/e2e_shadow_indexing_test.py

```diff
@@ -110,3 +123,602 @@ def test_write(self):
+
+        # Fetch the data. The data should go
+        # into the SI cache.
+        consumer.start()
+        consumer.wait()
+
+        assert len(consumer.messages) == msg_count
+
+        # The SI cache should have segments in it.
+        # Check if those segment files existed before retention
+        self.logger.debug('After fetch')
+        self._check_si_cache_after_fetch(data_dir_segments)
+
+    # Test the SI cache limit
+    @cluster(num_nodes=4)
+    def test_si_cache_limit(self):
+        # Node alloc is first because CI fails in the debug pipeline due to
+        # the warning message "Test requested X nodes, used only Y"
+        fgo_node = self.test_context.cluster.alloc(ClusterSpec.simple_linux(1))
+        self.logger.debug(f"Allocated verifier node {fgo_node[0].name}")
+
+        self.logger.info(f"Environment: {os.environ}")
+        if os.environ.get('BUILD_TYPE', None) == 'debug':
+            self.logger.info(
+                "Skipping test in debug mode (requires release build)")
+            return
+
+        # Remote write/read set at topic level
+        rpk = RpkTool(self.redpanda)
+        rpk.alter_topic_config(self.topic, 'redpanda.remote.write', 'true')
+        rpk.alter_topic_config(self.topic, 'redpanda.remote.read', 'true')
+        rpk.alter_topic_config(self.topic, 'retention.bytes',
+                               str(self.segment_size))
+
+        # 100k messages of size 2**18
+        # is ~24GB of data.
+        msg_size = 2**18
+        msg_count = 100000
+
+        self._gen_manifests(msg_size, fgo_node)
+
+        producer = FranzGoVerifiableProducer(self.test_context, self.redpanda,
+                                             self.topic, msg_size, msg_count,
+                                             fgo_node)
+
+        # The SI cache is set to 5GB in size. Test the cache limit by
+        # running 8 readers that consume 1GB segments effectively forcing
+        # ~8GB of IO. The readers read from random offsets.
+        rand_consumer = FranzGoVerifiableRandomConsumer(
+            self.test_context, self.redpanda, self.topic, msg_size, 10, 8,
+            fgo_node)
+
+        producer.start(clean=False)
+        rand_consumer.start(clean=False)
+
+        producer.wait()
+        rand_consumer.shutdown()
+        rand_consumer.wait()
+
+        # Return True if x is within the percent threshold of
+        # the given size
+        def in_percent_threshold(x, size, threshold):
+            percent_of_size = size * (threshold / 100)
+            percent_threshold = size + percent_of_size
+            self.logger.debug(f'Percent threshold: {percent_threshold}')
+            return x < percent_threshold
+
+        # Check SI cache size on all nodes
+        def check_si_cache_size():
+            is_cache_size_respected = True
+
+            for node in self.redpanda.nodes:
+                cmd = 'df --block-size=1 /var/lib/redpanda/data/cloud_storage_cache | awk \'{print $3}\' | tail -n1'
```

Comment:
I'm still not understanding the purpose of df for this test.
The original test is https://github.com/redpanda-data/redpanda/issues/3334

Reply:
maybe tag the author of the original test?

Reply:
The `df` lets us detect space consumed through inodes that aren't linked in to the filesystem tree (e.g. deleted files held open).

In the original situation on #3334 we were on servers with a dedicated filesystem for the cache, so the df and du numbers were directly comparable.  For the ducktape test you have to do a bit more work.

To get comparable values for du and df, you could take a baseline of both at the start of the test, and then subsequent compare the delta between the start of test and present: after writing in 1GB of data to an r=3 topic, you should see that both numbers go up by 3GB (to within some error margin to account for filesystem overhead in the df numbers).  If you see the du number go up by 3GB but the df number go up by 20GB, that's the red flag that something's getting leaked.

In the above the "du number" would be the sum of the data and cache directory sizes, and you may also need to account for the space consumed by the redpanda debug log file.  Longer term I would strongly prefer to get these tests nodes using dedicated XFS filesystems (https://github.com/redpanda-data/redpanda/issues/2376), at which point the cache and data dir can probably be on separate filesystems too.

Reply:
Thanks for the clarification @jcsp 
I'll make the necessary changes in the next push.